var/spack/repos/builtin/packages/mvapich2-gdr/package.py

```diff
@@ -113,6 +114,18 @@ def process_manager_options(self):
             opts.append("--enable-hip=basic")
             opts.append("--enable-rocm")
 
+        if "process_managers=mpiexec" in spec:
+            opts.append("--with-pm=mpiexec")
+            if "pmi_version=simple" in spec:
+                opts.append("--with-pmi=simple")
+            if "pmi_version=pmi1" in spec:
+                opts.append("--with-pmi=pmi1")
+            if "pmi_version=pmi2" in spec:
+                opts.append("--with-pmi=pmi2")
+            if "pmi_version=pmix" in spec:
+                opts.append("--with-pmi=pmix")
+                opts.append("--with-pmix={0}".format(spec["pmix"].prefix))
```

Comment:
Have you seen the `with_or_without` helper available for `AutotoolsPackage`s?  https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#with-or-without

Reply:
I am not familiar with this. @MatthewLieber please take a look and see if this would improve our packages.

Reply:
@MatthewLieber

Reply:
I do not believe the with_or_without worked well with our configure system but I have gone in and tryed to cut down on all of the appending of pmi options